src/index.js

```diff
@@ -335,10 +337,63 @@ class JanusAdapter {
     });
   }
 
+  toggleFreeze() {
+    if(this.frozen) {
+      this.unfreeze();
+    } else {
+      this.freeze();
+    }
+  }
+
+  freeze() {
+    this.frozen = true;
+  }
+
+  unfreeze() {
+    this.frozen = false;
+    this.flushPendingUpdates();
+  }
+
+  flushPendingUpdates() {
+    for (let [networkId, message] of this.frozenUpdates) {
+        if(!message.data.owner || this.occupants[message.data.owner]) { // ignore users who have disconnected since freezing, since their entities will have already been removed by NAF
+          this.onOccupantMessage(null, message.dataType, message.data);
+      }
+    }
+    this.frozenUpdates.clear();
+  }
+
+  storeMessage(message) {
+    const networkId = message.data.networkId;
+    if(!this.frozenUpdates.has(networkId)) {
+      this.frozenUpdates.set(networkId, message);
+    } else {
+      const storedMessage = this.frozenUpdates.get(networkId);
+
+      // Avoid updating components if the entity data received did not come from the current owner.
+      if (message.data.lastOwnerTime < storedMessage.data.lastOwnerTime ||
+            (storedMessage.data.lastOwnerTime === message.data.lastOwnerTime && storedMessage.data.owner > message.data.owner)) {
+        return;
+      }
+
+      // Delete messages override any other messages for this entity
+      if(message.dataType === "r") {
+        this.frozenUpdates.set(networkId, message);
+      } else {
+        // merge in component updates
+        Object.assign(storedMessage.data.components, message.data.components);
```

Comment:
This will not do a deep-merge, it will replace sub-properties. Is that okay?

Reply:
Yeah, I think we may want to use deep merge here.

Reply:
I originally had a deep merge but I am pretty sure components are always sent in full so we dont need to merge individual properties.

Reply:
Yeah, looks like it: https://github.com/networked-aframe/networked-aframe/blob/master/src/ComponentHelper.js#L4

Seems good then 👍